### ISSUE TITLE
cmake

### ISSUE THREAD

**Issue by swan46** [4lex4/scantailor-advanced#175]
I successfully built with cmake. I like scantailor advanced because of its speed. Thank you.
The part in cmake:
# Disable in-source builds.
#if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
#  message(FATAL_ERROR
#      "FATAL: In-source builds are not allowed."
#      " You should create a separate directory for build files.")
#endif()
I had to comment out. I think, this part is against the gist of cmake.
How can executable scanmake-advanced be downloaded and installed using apt-get?
Thanks again for the nice  scantailor version.

**Comment by derrikF** [comment]
AppImage - https://github.com/probonopd/scantailor-advanced/releases

**Comment by mara004** [comment]
Hmm, what the error message says is that you should create a separate build directory, as people usually do with cmake. Apparently you didn't?

**Comment by swan46** [comment]
I do not think, it is necessary to create a separate build
directory. I commented that check out, and it built
for me well. I would take that check out.

2021. 07. 12. 16:46 keltezéssel, mara004 írta:

**Comment by mara004** [comment]
As far as I know, most cmake projects do enforce a separate build directory. Building into the source tree is against good style.

**Comment by mara004** [comment]
> How can executable scanmake-advanced [sic] be downloaded and installed using apt-get?

What do you mean by this? Is it about installing ScanTailor Advanced from the distribution repository, or bundling a package from the compiled code?
Most distributions do not have STA available in their repositories, and they probably will not add it as it is unmaintained.
For bundling a package, you can use `cpack -G $target` (see https://github.com/4lex4/scantailor-libs-build#packaging).

**Comment by swan46** [comment]
If I were you, I would change that check to a warning, and
not leave it as an error. But it is your tree and your decision,
of course.
I know only trees, where the result gets into the source tree
somewhere.  And it is good like that. Therefore are branches
in a tree. No reason to force unusual things.

2021. 07. 14. 13:01 keltezéssel, mara004 írta:

**Comment by mara004** [comment]
> No reason to force unusual things.

Again, this is not at all unusual... I think the primary reason to explicitly forbid building into the source tree is to avoid compiled objects from being added/committed to git, and it just simplifies developer work if all source code files go in one directory and all compiled files go into another, so as to get a clean overview.


> If I were you

Who do you mean by `you`? Me? I don't maintain this software, neither do I have commit rights. I'm only a user.

**Comment by swan46** [comment]
I see.
That check causes, that users are not able to build
scantailor advanced. Not in git, but on their own
computer. Which is a pity.

2021. 07. 15. 13:20 keltezéssel, mara004 írta:

**Comment by mara004** [comment]
> That check causes, that users are not able to build
> scantailor advanced.

Not at all. Just use a separate build directory as you should, and it will work perfectly well.

**Comment by swan46** [comment]
You seem to be a troll for me.

2021. 07. 16. 11:24 keltezéssel, mara004 írta:

**Comment by mara004** [comment]
And you to me... What is there to be gained from spamming the issue tracker of an abandoned project with comments that are obviously wrong? If you look around, the vast majority of cmake projects either recommend or enforce a separate build directory.

**Comment by mara004** [comment]
I have blocked and reported you now.
